### ISSUE TITLE
Rule request: identical operands in comparison operator rule

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1371]
```swift
// not ok
rhs.applyToDictionaries == rhs.applyToDictionaries

// ok
lhs.applyToDictionaries == rhs.applyToDictionaries
```

See #1368 for more history.

**Comment by masters3d** [comment]
What do you think of this regex? `(\b\S*\b)\s*(==|!=|<|>|=<|=>)\s*\1`
escaped: `"(\\b\\S*\\b)\\s*(==|!=|<|>|=<|=>)\\s*\\1"`
![image](https://cloud.githubusercontent.com/assets/6539412/26077559/583dd60a-3971-11e7-844e-5b281b9f82fd.png)

**Comment by masters3d** [comment]
I think instead of matching anything `\S` we should probably match only valid operands `[A-Za-z0-9_\.]+`
That would give us this:
`(\b[A-Za-z0-9_\.]+\b)\s*(==|!=|<|>|<=|>=)\s*\1`

**Comment by masters3d** [comment]
@marcelofabri Do you think this rule should be on by default?

**Comment by jpsim** [comment]
Seems like a reasonable place to start to me! OSSCheck would undoubtedly catch if this produces common false positives.

**Comment by masters3d** [comment]
@jpsim You had an idea of more rules around this sort of line of thinking? (I think I remember you posted something on twitter with a document that pointed out some of these common issues.)

**Comment by masters3d** [comment]
https://www.viva64.com/en/b/0509/

**Comment by ZevEisenberg** [comment]
Just lost a half hour to this exact issue. Came here to post my +1.

**Comment by masters3d** [comment]
```
Equal to (a == a)
Not equal to (a != a)
Greater than (a > a)
Less than (a < a)
Greater than or equal to (a >= a)
Less than or equal to (a <= a)
```
In addition to the basic operators above

we should also include:
Logical AND (a && b)  // avoid `a && a `
Logical OR (a || b) // avoid `a || a` 

surprisingly Logical NOT (!a) can be essentially ignored which is good since this could have been tripped the unwrap operator.

ternary conditional operator:
`bool ? a :  a` // Avoid same value for true and false case.
